src/server/middleware/auth.ts

```diff
@@ -154,6 +154,13 @@ const requireEditAssessmentFile = async (req: Request, _res: Response, next: Nex
   }
 }
 
+const requireGetAreas = async (req: Request, _res: Response, next: NextFunction) => {
+  const { assessmentName, cycleName } = <Record<string, string>>{ ...req.params, ...req.query, ...req.body }
+  const user = Requests.getRequestUser(req)
+  if (!user) _next(!assessmentName && !cycleName, next)
+  else _next(!!user, next)
```

Comment:
When Assessment and cycle are specified , it's needed to check that the user has read view permission to them.

The condition should be if not Assessment and not cycle then _next else requireView

Reply:
yes, but if cycle is not published, and user is logged then user must have an active role in that cycle to be authorized to view